internal/schema/custom_nested_object_test.go

```diff
@@ -1375,12 +1455,22 @@ AttrTypes: v.ObjectAttribute.AttributeTypes(ctx),
 }), diags
 }
 
-objVal, diags := types.ObjectValue(
-map[string]attr.Type{
+attributeTypes := map[string]attr.Type{
```

Comment:
I just noticed, that there's also `AttributeTypes()` defined returning this map – do you think we could re-use that too? 😄🤔

Reply:
Good catch! Let's do that

Reply:
After further checking the implementation here:
```
attributeTypes := map[string]attr.Type{
{{- range $key, $value := .AttributeTypes }}
{{- if eq $value "Object"}}
"{{$key}}": basetypes.ObjectType{
AttrTypes: v.{{$key.ToPrefixPascalCase $.Name}}.AttributeTypes(ctx),
},
{{- else}}
"{{$key}}": {{index $.AttrTypes $key}},
{{- end}}
{{- end}}
}
```
differs from `AttributeTypes()` (which does not recurse one level into objects):
```
func (v {{.Name}}Value) AttributeTypes(ctx context.Context) map[string]attr.Type {
return map[string]attr.Type{
{{- range $key, $value := .AttrTypes }}
"{{$key}}": {{$value}},
{{- end}}
}
}
```
While this is probably a bug, I decided against going down that rabbit hole as part of this PR 😄 So I'm going to resolve this discussion.